core/src/main/scala/akka/kafka/scaladsl/SendProducer.scala

```diff
@@ -96,6 +96,10 @@ final class SendProducer[K, V] private (val settings: ProducerSettings[K, V])(im
 }
 
 object SendProducer {
-  def apply[K, V](settings: ProducerSettings[K, V])(implicit system: ActorSystem): SendProducer[K, V] =
-    new SendProducer(settings)
+  def apply[K, V](settings: ProducerSettings[K, V])(implicit system: ClassicActorSystemProvider): SendProducer[K, V] =
```

Comment:
It's interesting that this remains binary compatible (according to MiMa).  I did a little experimentation with overloaded methods that contain different implicit parameter types and discovered the magnet pattern solution in [this SO answer](https://stackoverflow.com/a/17366915/895309). I tried implementing it in a branch and it breaks binary compatibility, but seems code compatibility.

https://github.com/akka/alpakka-kafka/compare/master...seglo:seglo/allow-classic-provider#diff-02cd8e6bc99d9aaa7e2ebdbe16e9d121R98-R114

Since your solution appears to satisfy both it seems like the right approach to me.